### DISCUSSION TITLE
Support for Multiple Dates Selection in DatePicker

### DISCUSSION THREAD

**Discussion by dimasvm** [filamentphp/filament#15167] [Ideas]
### **Request Feature:**
Currently, the DatePicker component only allows for single date selection. In scenarios where users need to select multiple non-consecutive dates (e.g., scheduling, event planning, or marking holidays), this limitation can be challenging. A built-in feature for multiple date selection would greatly enhance flexibility and usability.

### **Solution**
Add support for a "multiple mode" in the DatePicker component, allowing users to select multiple dates. Ideally, the implementation could include:

A multiple property (e.g., multiple: true).
The ability to store selected dates as an array of strings or Date objects.
Customizable behavior for the display, such as showing all selected dates in the input field or a dropdown list.

Example usage:
```
DatePicker::make('selected_dates')  
    ->multiple()
    ->label('Select Dates')  
    ->placeholder('Pick one or more dates')
```

### **Alternatives:**
An alternative is to use multiple instances of DatePicker or custom implementations using other JavaScript libraries like Flatpickr. However, these approaches can complicate integration and styling consistency within Filament.

### **Additional context**
This feature could be valuable in various use cases, such as:

Booking systems where users need to select multiple dates for reservations.
Scheduling applications where multiple events occur on non-consecutive days.
Marking specific days as holidays or important dates.
Thank you for considering this request! Let me know if you need more details or examples.

**Comment by jamiemcd** [comment]
I added my upvote. One solution for now might be using the Calendar component that comes with Flux Pro. It supports multiple non-consecutive date selection.

**Comment by matildasmets** [comment]
+1
Also a similar idea of mine, but for ranges instead of multiple:
https://github.com/filamentphp/filament/discussions/18902
